python/datafusion/dataframe.py

```diff
@@ -160,6 +160,9 @@ def with_column(self, name: str, expr: Expr) -> DataFrame:
         """
         return DataFrame(self.df.with_column(name, expr.expr))
 
+    @deprecated(
+        "with_column_renamed() is deprecated. Use :py:meth:`~DataFrame.rename` instead"
+    )
```

Comment:
I would remove this `deprecated` marker. I think it's okay to have this kind of alias permanently as it is useful to end users.